compiler/aarch64/codegen/OMRCodeGenerator.cpp

```diff
@@ -574,6 +576,54 @@ int64_t OMR::ARM64::CodeGenerator::getSmallestPosConstThatMustBeMaterialized()
    return 0;
    }
 
+
+bool OMR::ARM64::CodeGenerator::getSupportsOpCodeForAutoSIMD(TR::ILOpCode opcode, TR::DataType dt)
+   {
+   // implemented vector opcodes
+   switch (opcode.getOpCodeValue())
+      {
+      case TR::vadd:
+      case TR::vsub:
+         if (dt == TR::Int64)
+            return true;
+         /* other types -- fall through */
```

Comment:
This is not easy to read -- Too many "fall through" cases.  I would suggest changing it like this:

```
case TR::vadd:
case TR::vsub:
   if (dt == TR::Int8 || dt == TR::Int16 || dt == TR::Int32 || dt == TR::Int64 || dt == TR::Float || dt == TR::Double)
      return true;
   else
      return false;
case TR::vmul:
   if (dt == TR::Int8 || dt == TR::Int16 || dt == TR::Int32 || dt == TR::Float || dt == TR::Double)
      return true;
   else
      return false; // Int64 is not supported
case TR::vdiv:
   if (dt == TR::Float || dt == TR::Double)
      return true;
   else
      return false; // Int8/Int16/Int32/Int64 are not supported
```

Reply:
Thank you for the feedback, I also felt the same. I will make this change.